pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -137,20 +150,25 @@ func (r *repository) Blobs(ctx context.Context) distribution.BlobStore {
 	repo := repository(*r)
 	repo.ctx = ctx
 
-	bs := &quotaRestrictedBlobStore{
-		BlobStore: r.Repository.Blobs(ctx),
-		repo:      &repo,
-	}
-	if !r.pullthrough {
-		return bs
-	}
+	bs := r.Repository.Blobs(ctx)
 
-	return &pullthroughBlobStore{
+	bs = &quotaRestrictedBlobStore{
 		BlobStore: bs,
 
-		repo:          &repo,
-		digestToStore: make(map[string]distribution.BlobStore),
+		repo: &repo,
 	}
+
+	if r.pullthrough {
```

Comment:
I'll leave it here to hang ;) Do we care about quota when doing pullthrough? I don't think so, from what you've told me we don't store anything doing so, why adding that unnecessary layer here? Or am I missing something here?